pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/PinotFS.java

```diff
@@ -40,32 +40,41 @@
   /**
    * Deletes the file at the location provided. If the segmentUri is a directory, it will delete the entire directory.
    * @param segmentUri URI of the segment
+   * @param forceDelete true if we want the uri and any suburis to always be deleted, true if we want delete to fail
+   *                    when we want to delete a directory and that directory is not empty
    * @return true if delete is successful else false
    * @throws IOException IO failure
+   * @throws Exception if segmentUri is not valid or present
    */
-  public abstract boolean delete(URI segmentUri) throws IOException;
+  public abstract boolean delete(URI segmentUri, boolean forceDelete) throws IOException;
 
   /**
    * Moves the file from the src to dst. Does not keep the original file. If the dst has parent directories
    * that haven't been created, this method will create all the necessary parent directories. If dst already exists,
    * it will overwrite. Will work either for moving a directory or a file. Currently assumes that both the srcUri
-   * and the dstUri are of the same type (both files or both directories).
+   * and the dstUri are of the same type (both files or both directories). If srcUri is a file, it will move the file to
+   * dstUri's location. If srcUri is a directory, it will move the directory to dstUri. Does not support moving a file under a
+   * directory.
    * @param srcUri URI of the original file
    * @param dstUri URI of the final file location
    * @param overwrite true if we want to overwrite the dstURI, false otherwise
    * @return true if move is successful
    * @throws IOException on failure
+   * @throws Exception if srcUri is not present or valid
    */
   public abstract boolean move(URI srcUri, URI dstUri, boolean overwrite) throws IOException;
 
   /**
    * Copies a file from src to dst. Keeps the original file. If the dst has parent directories that haven't
    * been created, this method will create all the necessary parent directories. If dst already exists, it will overwrite.
-   * Works both for moving a directory and a file. Assumes that srcUri and dstUri are of the same type (both files and directories)
+   * Works both for copying a directory and a file. If srcUri is a file, it will copy the file to
+   * dstUri's location. If srcUri is a directory, it will copy the directory to dstUri. Does not support copying a file under a
```

Comment:
Not clear if source dir is a/b/c, and destdir is x/y/z, then will the copied dir be x/y/z/c or x/y/z (overwriting 'z' if it exists).